Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -3149,6 +3176,20 @@ template <class TreeBuilder> TreeClassExpression Parser<LexerType>::parseClass(T
                 property = context.createProperty(ident, computedPropertyName, initializer, type, SuperBinding::NotNeeded, tag);
             else
                 property = context.createProperty(ident, initializer, type, SuperBinding::NotNeeded, inferName, tag);
+        } else if (parseMode == SourceParseMode::ClassStaticBlockMode) {
+            dataLogIf(Options::useDebugLog(), "-------- parseClass ClassStaticBlockMode case start\n");
+            ParserFunctionInfo<TreeBuilder> methodInfo;
+            methodInfo.name = ident;
+
+            SetForScope innerParseMode(m_parseMode, parseMode);
+            failIfFalse((parseFunctionInfo(context, FunctionNameRequirements::Unnamed, false, ConstructorKind::None, SuperBinding::Needed, methodStart, methodInfo, FunctionDefinitionType::StaticBlock)), "Cannot parse this static block");
+
+            TreeExpression method = context.createMethodDefinition(methodLocation, methodInfo);
+            property = context.createProperty(methodInfo.name, method, type, SuperBinding::Needed, InferName::Allowed, tag);
+
+            // property = context.createProperty(ident, type, SuperBinding::Needed, InferName::Allowed, tag);
+            
+            dataLogIf(Options::useDebugLog(), "-------- parseClass ClassStaticBlockMode case end\n");
```

Comment:
We need to consume the tokens inside static block and create a property for the static block.

Reply:
Currently, I just parse it as a `FuncExprNode`. We could parse it as `BlockNode`. So, `FuncExprNode` vs `BlockNode`?

Reply:
How can we reuse the ASTNode (`FuncExprNode ` or `BlockNode `) in parseClassFieldInitializerSourceElements?